rfcs/45-stageless.md

```diff
@@ -0,0 +1,866 @@
+causing the **scheduler** to run the systems in parallel using a strategy that respects all of the configured constraints.
+If these constraints cannot be met (for example, a system may want to run both before and after another system), the schedule is said to be **unsatisfiable**, and the scheduler will panic.
+Schedules can be nested and branched from within **exclusive systems** (which have mutable access to the entire `World`), allowing you to encode arbitrarily complex control flow using ordinary Rust constructs.
+
+By default, each new `Schedule` is entirely unordered: systems will be selected in an arbitrary order (which is not guaranteed to be stable across passes of the schedule) and run if and only  if all of the data that they must access are free.
+Just like with standard borrow checking, multiple systems can read from the same data at once, but writing to the data requires an exclusive lock.
+Systems which cannot be run in parallel are said to be **incompatible**.
+
+In Bevy, schedules are ordered via **intent-based configuration**: record the exact, minimal set of constraints needed for a scheduling strategy to be valid, and then let the scheduler figure it out!
+As the number of systems in your app grows, the number of possible scheduling strategies grows exponentially, and the set of *valid* scheduling strategies grows almost as quickly.
+However, each system will continue to work hand-in-hand with a small number of closely-related systems, interlocking in a straightforward and comprehensible way.
+
+By configuring our systems with straightforward, local rules, we can allow the scheduler to pick one of the possible valid paths, optimizing as it sees fit without breaking our logic.
+Just as importantly, we are not over-constraining our ordering. Subtle ordering bugs will surface quickly and then be stamped out with a well-considered rule.
+
+### The `App` stores multiple schedules
+
+The `App` can store multiple `Schedules`: each with their own `impl ScheduleLabel` type, allowing you to cleanly execute coherent blocks of logic when the need arises.
+By default, each app stores both a startup and main schedule: the former runs only once on app creation, while the latter loops endlessly.
+However, schedules are a surprisingly powerful and flexible tool: they can be used to handle initialization and cleanup logic when working with states or used ad-hoc to run game logic in complex patterns.
+
+The main, startup and rendering schedules can be accessed using the `DefaultSchedule::Main`, `DefaultSchedule::Startup`, `DefaultSchedule::Rendering` labels respectively.
+By default, systems are added to the main schedule.
+You can control this by adding the `.to_schedule(ScheduleLabel::Variant)` system descriptor to your system.
+To add an entirely new schedule to your app (which can be run on the world in an exclusive system), use `app.add_schedule(label, schedule)`.
+

```

Comment:
`Startup`, `Main` and `Rendering` are listed together,
but `Rendering` is nested inside of `Main` as an exclusive system.
This can be confusing with `add_schedule(label, schedule)`,
where our schedules are nested inside of `Main` in the same fashion `Rendering` is.